common/compositor/va/varenderer.cpp

```diff
@@ -440,18 +448,33 @@ bool VARenderer::UpdateCaps() {
       if (use_default) {
         itr->second.value_ = itr->second.caps_.range.default_value;
       }
-      cbparam.type = VAProcFilterColorBalance;
-      cbparam.value = itr->second.value_;
-      cbparam.attrib = itr->second.caps_.type;
-      if (!cb_elements[static_cast<int>(itr->first)].CreateBuffer(
-              va_context_, VAProcFilterParameterBufferType,
-              sizeof(VAProcFilterParameterBufferColorBalance), 1, &cbparam)) {
-        return false;
-      }
-      filters_.push_back(cb_elements[static_cast<int>(itr->first)].buffer());
+      cbparam[index].type = VAProcFilterColorBalance;
+      cbparam[index].value = itr->second.value_;
+      cbparam[index].attrib = itr->second.caps_.type;
+      index++;
     }
   }
 
+  // workarround if filter num is 0, then set HSBC/Deinterlace will meet green
+  // screen.
+  if (!index) {
```

Comment:
Just for reminder, this may have performance issue, since it always set the hue. @feiwan1 . please reproduce the issue using libyami, we report to driver team and fix the green screen issue in driver.

Reply:
Can we just disable setting HSBC in this case and fix the driver issue. Dont want to take the W/A here

Reply:
Currently, we find if we do not set the filter, and then after some frame later, we set the filter, it will turn green. 
Yeah, I do not like this too, but it takes time to fix this in the driver. 
Is it possible to merge this first, to unblock this milestone.
then we will fix this in the driver ASAP

Reply:
Let's fix this in driver @xuguangxin, otherwise these things get lost easily. Milestone pressure gets things fixed quickly :)

Reply:
When set param_.filter_flags as none-zero, the issue will disappear. But not sure why this happens, no va error returns, no hwc error.